### ISSUE TITLE
govet issues not reported

### ISSUE THREAD

**Issue by egurnov** [golangci/golangci-lint#485]
Go vet issues on MacOS are not reported, however the same code/setup reports on Linux (Docker container). I will provide MVE setup for both.

Here's my example code:
```go
package main

import "fmt"

func main() {
        var p *int
        if p == nil && *p == 0 {
                fmt.Println("Too bad")
        }
}
```

### MacOS Mojave 10.14.3
1. Version of golangci-lint:
I used brew, so `--version` flag is not useful:
```console
$ golangci-lint --version
Error: unknown flag: --version
```
But here's brew info output:
```console
$ brew info golangci-lint
golangci/tap/golangci-lint: stable 1.16.0
Fast linters runner for Go.
https://golangci.com
/usr/local/Cellar/golangci-lint/1.16.0 (5 files, 21.9MB) *
  Built from source on 2019-04-02 at 16:03:36
From: https://github.com/golangci/homebrew-tap/blob/master/Formula/golangci-lint.rb
```
2. Config file: no config
3. Go environment:
```console
$ go version && go env
go version go1.11.4 darwin/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/alexanderegurnov/Library/Caches/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/alexanderegurnov/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/Cellar/go/1.11.4/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.11.4/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/66/s74987090sj54m_jkt801mf00000gn/T/go-build076108946=/tmp/go-build -gno-record-gcc-switches -fno-common"
```
4. Verbose output of running: `golangci-lint run -v`
```console
$ golangci-lint run -v --no-config --disable-all --enable=govet --exclude-use-default=false
INFO [lintersdb] Active 1 linters: [govet]
INFO [loader] Go packages loading at mode load types and syntax took 239.814379ms
INFO [runner] worker.2 took 18.089µs
INFO [runner] worker.4 took 2.636µs
INFO [runner] worker.3 took 65.403µs
INFO [runner] worker.1 took 3.537767ms with stages: govet: 3.506322ms
INFO [runner] Workers idle times: #2: 3.258434ms, #3: 2.891923ms, #4: 3.181715ms
INFO [runner] processing took 20.835µs with stages: max_same_issues: 16.539µs, skip_dirs: 522ns, nolint: 441ns, path_prettifier: 428ns, max_from_linter: 408ns, skip_files: 338ns, autogenerated_exclude: 270ns, diff: 259ns, identifier_marker: 208ns, exclude-rules: 207ns, cgo: 206ns, max_per_file_from_linter: 203ns, replacement_builder: 200ns, exclude: 196ns, source_code: 145ns, uniq_by_line: 133ns, path_shortener: 132ns
INFO Memory: 4 samples, avg is 68.9MB, max is 68.9MB
INFO Execution took 351.795401ms
```

### Linux (`docker run golangci/golangci-lint:v1.16.0`):
1. Version: `docker pull golangci/golangci-lint:v1.16.0`
```console
$ docker run golangci/golangci-lint:v1.16.0 golangci-lint --version
golangci-lint has version 1.16.0 built from 97ea1cb on 2019-03-31T19:48:46Z
```
2. Config file: no config
3. Go environment:
```console
$ docker run golangci/golangci-lint:v1.16.0 /bin/bash -c 'go version && go env'
go version go1.12.1 linux/amd64
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/go"
GOPROXY=""
GORACE=""
GOROOT="/usr/local/go"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build327444595=/tmp/go-build -gno-record-gcc-switches"
```
4. Verbose output of running: `golangci-lint run -v`
```console
$ docker run  \
>   --volume "$(pwd):/go/src/github.com/egurnov/mve/vettest" \
>   --workdir "/go/src/github.com/egurnov/mve/vettest" \
>   golangci/golangci-lint:v1.16.0 \
>   golangci-lint run -v \
>     --no-config \
>     --disable-all \
>     --enable=govet \
>     --exclude-use-default=false
level=info msg="[lintersdb] Active 1 linters: [govet]"
level=info msg="[loader] Go packages loading at mode load deps types and syntax took 945.2639ms"
level=info msg="[loader] SSA repr building timing: packages building 9.0155ms, total 119.826ms"
level=info msg="[runner] worker.2 took 9.7µs"
level=info msg="[runner] worker.1 took 96.3356ms with stages: govet: 96.2807ms"
level=info msg="[runner] Workers idle times: #2: 84.5987ms"
main.go:7:7: nilness: tautological condition: nil == nil (govet)
	if p == nil && *p == 0 {
	     ^
level=info msg="[runner] Issues before processing: 2, after processing: 1"
level=info msg="[runner] processing took 7.2414ms with stages: source_code: 3.0562ms, cgo: 1.1905ms, skip_dirs: 918.1µs, nolint: 573.9µs, autogenerated_exclude: 506.9µs, path_prettifier: 377.6µs, max_same_issues: 367.6µs, identifier_marker: 69.2µs, path_shortener: 38.5µs, replacement_builder: 28.9µs, uniq_by_line: 18.8µs, max_from_linter: 18.1µs, diff: 16.3µs, exclude-rules: 15.5µs, max_per_file_from_linter: 15.3µs, exclude: 15.3µs, skip_files: 14.7µs"
level=info msg="File cache stats: 1 entries of total size 110B"
level=info msg="Memory: 14 samples, avg is 121.1MB, max is 202.3MB"
level=info msg="Execution took 1.2315556s"
```

**Comment by jirfag** [comment]
Hi!
1. It works ok on my mac os
2. I guess you are using compiled from source golangci-lint because brew-installed golangci-lint supports `--version` flag

**Comment by egurnov** [comment]
Hi @jirfag!

You are right, as my brew output says, `Built from source on 2019-04-02 at 16:03:36`. However this is the result of running `brew install golangci-lint`.

Meanwhile I've found the opposite kind of error: MacOS version finds what Docker version doesn't see. Running with the same versions as above.

### MVE
https://github.com/egurnov/mve/tree/master/golinttest-min

main.go
```go
package main

import "github.com/egurnov/mve/golinttest-min/other"

func main() {
	var x int64 = other.MakeInt64(6)
	_ = x
}
```

other/other.go
```go
package other

// MakeInt64 func
func MakeInt64(x int) int64 {
	return int64(x)
}
```

#### MacOS Mojave 10.14.3
```console
$ golangci-lint run ./... -v --no-config --disable-all --enable=govet,golint --exclude-use-default=false
INFO [lintersdb] Active 2 linters: [golint govet]
INFO [loader] Go packages loading at mode load types and syntax took 257.299328ms
INFO [runner] worker.3 took 17.401µs
INFO [runner] worker.1 took 1.339µs
INFO [runner] worker.2 took 3.075829ms with stages: golint: 3.050909ms
main.go:6:8: should omit type int64 from declaration of var x; it will be inferred from the right-hand side (golint)
	var x int64 = other.MakeInt64(6)
	      ^
INFO [runner] worker.4 took 4.763047ms with stages: govet: 4.739004ms
INFO [runner] processing took 912.866µs with stages: identifier_marker: 321.774µs, autogenerated_exclude: 283.629µs, path_prettifier: 175.15µs, source_code: 59.683µs, cgo: 23.344µs, nolint: 16.832µs, skip_dirs: 15.117µs, max_same_issues: 8.417µs, uniq_by_line: 3.581µs, path_shortener: 1.146µs, max_from_linter: 1.075µs, exclude-rules: 676ns, replacement_builder: 655ns, max_per_file_from_linter: 615ns, diff: 569ns, skip_files: 340ns, exclude: 263ns
INFO [runner] Workers idle times: #1: 3.990167ms, #2: 1.680441ms, #3: 4.004054ms
INFO Memory: 5 samples, avg is 68.9MB, max is 68.9MB
INFO Execution took 472.04612ms
```
#### Docker
```console
$ docker run    --volume "$(pwd):/go/src/github.com/egurnov/mve/golinttest-min" --workdir "/go/src/github.com/egurnov/mve/golinttest-min"   golangci/golangci-lint:v1.16.0   golangci-lint run ./... -v     --no-config     --disable-all     --enable=govet,golint     --exclude-use-default=false
level=info msg="[lintersdb] Active 2 linters: [golint govet]"
level=info msg="[loader] Go packages loading at mode load deps types and syntax took 208.0059ms"
level=info msg="[loader] SSA repr building timing: packages building 171.3µs, total 361.3µs"
level=info msg="[runner] worker.2 took 1.2716ms with stages: govet: 1.2424ms"
level=info msg="[runner] worker.1 took 6.6069ms with stages: golint: 6.1606ms"
level=info msg="[runner] Workers idle times: #2: 5.1329ms"
level=info msg="[runner] processing took 262.3µs with stages: skip_dirs: 73µs, max_same_issues: 46.2µs, identifier_marker: 26.2µs, max_from_linter: 10.2µs, nolint: 10.1µs, cgo: 8.7µs, max_per_file_from_linter: 8.3µs, replacement_builder: 8.2µs, skip_files: 8.2µs, uniq_by_line: 8.2µs, path_prettifier: 8µs, exclude-rules: 8µs, exclude: 7.9µs, source_code: 7.8µs, autogenerated_exclude: 7.8µs, diff: 7.8µs, path_shortener: 7.7µs"
level=info msg="File cache stats: 0 entries of total size 0B"
level=info msg="Memory: 4 samples, avg is 68.4MB, max is 68.4MB"
level=info msg="Execution took 287.74ms"
```
